File: src/main/java/com/vk/lingvobot/entities/menu/MenuLevel.java
package com.vk.lingvobot.entities.menu;

public enum MenuLevel {
    MAIN("MAIN"),
    DIALOG("DIALOG"),
    PHRASE("PHRASE"),
    PHRASE_RUS_ENG("PHRASE_RUS_ENG"),
    PHRASE_ENG_RUS("PHRASE_ENG_RUS"),
    IMPORT_DIALOG("IMPORT_DIALOG");

    private final String code;

    public MenuLevel fromCode(String code) {
        for (MenuLevel decisionType: values()) {
            if (decisionType.code.equals(code)) {
                return decisionType;
            }
        }
        throw new UnsupportedOperationException("The code " + code + " is not supported!");
    }

    MenuLevel(String code) {
        this.code = code;
    }
}

File: src/main/kotlin/com/vk/lingvobot/entities/menu/MenuLevel.kt
package com.vk.lingvobot.entities.menu

enum class MenuLevel {
    MAIN("MAIN"),
    DIALOG("DIALOG"),
    PHRASE("PHRASE");

    private val code: String

    constructor(code: String) {
        this.code = code
    }

    open fun fromCode(code: String): MenuLevel? {
        for (decisionType in values()) {
            if (decisionType.code == code) {
                return decisionType
            }
        }
        throw UnsupportedOperationException("The code $code is not supported!")
    }
}